test/integration/mongodb-handshake/mongodb-handshake.prose.test.ts

```diff
@@ -110,6 +116,49 @@ describe('Handshake Prose Tests', function () {
     });
   }
 
+  context('Test 9: Valid container and FaaS provider', function () {
+    const env = [
+      ['AWS_EXECUTION_ENV', 'AWS_Lambda_java8'],
+      ['AWS_REGION', 'us-east-2'],
+      ['AWS_LAMBDA_FUNCTION_MEMORY_SIZE', '1024'],
+      ['KUBERNETES_SERVICE_HOST', '1']
+    ] as EnvironmentVariables;
+    before(() => {
+      for (const [key, value] of env) {
+        process.env[key] = value;
+      }
+    });
+    after(() => {
+      for (const [key] of env) {
+        delete process.env[key];
+      }
+    });
```

Comment:
```suggestion
      let beforeEnv;
      before(() => {
        beforeEnv = new Map(Object.entries(process.env));
        for (const [key, value] of env) {
          process.env[key] = value;
        }
      });
      
      after(() => {
        for (const [key] of env) {
          if (beforeEnv.has(key)) {
            process.env[key] = beforeEnv.get(key);
          } else {
            delete process.env[key];
          }
        }
      });
```

Currently, we're deleting env vars that may have been present before the test ones were set. Most of these env vars aren't important to any other testing but I am suspicious of `AWS_REGION` and how that could impact other tests if not today, in the future.

Reply:
This is how the other tests in this file work. Do you want me to update the rest of them as well?